### ISSUE TITLE
Addressing points rather than extents

### ISSUE THREAD

**Issue by tomcrane** [IIIF/api#1387]
IIIF (via OA/W3C annotations) uses the [W3C Media Fragments specification](https://www.w3.org/TR/media-frags/) to target spatial and temporal extents of the canvas:

`xywh=500,500,100,100`

`t=10,20`

A common annotation use case is map markers or other point annotations. A common solution to this today is a target that looks like `500,500,1,1`. For time-based media, points might be used to indicate a cut in a film, or where to start playing a clip from.

The media fragments spec has no support for a point, e.g., `xy=500,500`. The syntax does allow for `t=20`, but says that it should be interpreted as `t=20,end`. We don't want the spatial equivalent of that, where our `xy` target is interpreted as "extend to the bottom right corner".

What should IIIF do about point targeting? Carry on using `500,500,1,1`? If so, this pattern should be documented. If not, something more elegant, to support `xy=500,500`, should go in the spec. We would then need to define what `t=20` means.

**Comment by zimeon** [comment]
If we do decide we really need a point then I think the clearest choices are either:

  1. use illegal zero-size fragments `t=10,10`, ’xywh=10,20,0,0` that otherwise follow the media fragments spec; or
  2. invent new point types along the lines of `tp=10`, `xy=10,20` that follow the start/top-left semantics from the media fragments spec

I do not think that redefining the meaning of `t=10` is a viable option because that would be a conflict with valid semantics in media fragments

**Comment by workergnome** [comment]
I don't see a use for regions of time with zero duration, nor regions of space with zero volume.  Even a map marker occurs at a single pixel, not the top-left corner of a pixel, and a frame of video has duration, even if it's just the duration of a single frame.  

Points (and instants) are really nasty things that are also imaginary.

**Comment by tomcrane** [comment]
Being imaginary doesn't stop us using latitude and longitude coordinates to put markers on a map, or talking about music using both instants and extents:

"the trombones enter at 4:14" vs "listen for the cowbells from 7:34-11:01"  

The distinction between targeting a point and targeting a region when annotating spatial or temporal content seems important to me. It's not about precision, it's about intent, and if the intent is clear from unambiguous syntax, a client can produce a different appropriate user experience. Pins rather than polygons, perhaps. Or ▼ on a timeline rather than ｜.

For spatial, `1,1` is currently a de facto magic syntax to indicate a point rather than a rectangle below the resolution of the canvas coordinates. I agree that people could live with that (although sometimes they use 0,0).

For time-based media there may or may not be a frame, but the canvas duration is unaware of that notion, it deals only in seconds. If I want to talk about various markers in a film or piece of music, annotation using time points as targets is much cleaner and easier for both humans and client software to understand, rather than arbitrary small time fragments. Clients already process points and use them to draw things on the canvas or timeline - points are the start and end of regions.

**Comment by azaroth42** [comment]
XY point selector for spatial selection:
```
{
  "start": {
    "type": "SpecificResource",
    "source": {
      "id": "https://example.org/iiif/id1/canvas1",
      "type": "Canvas"
    },
    "selector": {
      "type": "PointSelector",
      "conformsTo": "http://iiif.io/api/annex/annotation/pointSelector/",
      "value" : "xy=10,10"
  }
}
```

**Comment by azaroth42** [comment]
Alternative:

```
{
  "start": {
    "type": "SpecificResource",
    "source": {
      "id": "https://example.org/iiif/id1/canvas1",
      "type": "Canvas"
    },
    "selector": {
      "type": "PointSelector",
      "conformsTo": "http://iiif.io/api/annex/annotation/pointSelector/",
      "x" : 10,
      "y": 10,
      "tp": 14.5
  }
}
```

**Comment by brion** [comment]
The separate-parameters form is less ambiguous with the media fragment selector form, which is good.

**Comment by azaroth42** [comment]
Per https://github.com/IIIF/api/issues/1332#issuecomment-369077215, I think the definition goes in the Annex and examples are used in the spec.

**Comment by zimeon** [comment]
Per https://github.com/IIIF/api/pull/1474#discussion_r171286663 I think it would be useful to add example of both full canvas reference and point-in-time reference in description of [start](http://prezi3.iiif.io/api/presentation/3.0/#start) (as opposed to current example which has a time-period fragment misused as time point...)

**Comment by azaroth42** [comment]
I don't think we need `conformsTo` in the examples above.  We don't say, for example, that the Image API Selector conforms to the Image API.  `conformsTo` would only be needed when there's multiple possible interpretations of x, y or t properties (e.g. as per URI fragments).

**Comment by azaroth42** [comment]
Closed  by #1500
